vaadin-app-layout-flow-parent/vaadin-app-layout-flow-integration-tests/src/test/java/com/vaadin/flow/component/applayout/test/AppLayoutI18nIT.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2000-2019 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.vaadin.flow.component.applayout.test;
+
+import com.vaadin.flow.component.applayout.testbench.AppLayoutElement;
+import com.vaadin.tests.AbstractParallelTest;
+
+import org.openqa.selenium.By;
+
+import org.junit.Test;
+import org.junit.Assert;
+import org.junit.Before;
+
+public class AppLayoutI18nIT extends AbstractParallelTest {
+    private AppLayoutElement layout;
+
+    @Before
+    public void init() {
+        String url = getBaseURL().replace(super.getBaseURL(),
+                super.getBaseURL() + "/vaadin-app-layout/i18n");
+        getDriver().get(url);
+        layout = $(AppLayoutElement.class).waitForFirst();
+    }
```

Comment:
Not sure why this pattern of building the URL and then using it with the driver exists, but there is an easier way:
- use the `@TestPath` annotation to specify the URL: https://github.com/vaadin/flow-components/blob/master/vaadin-menu-bar-flow-parent/vaadin-menu-bar-flow-integration-tests/src/test/java/com/vaadin/flow/component/menubar/tests/MenuBarPageIT.java#L31
- then just call `open()` in the init method: https://github.com/vaadin/flow-components/blob/master/vaadin-menu-bar-flow-parent/vaadin-menu-bar-flow-integration-tests/src/test/java/com/vaadin/flow/component/menubar/tests/MenuBarPageIT.java#L40

That might require extending from `AbstractComponentIT` instead, but `AbstractParallelTest` is probably not needed here.

Reply:
I absolutely agree. However, I tried to use `AbstractComponentIT` and it didn't work failing with the error:

```
java.lang.IllegalStateException: The driver executable does not exist: /Users/vursen/dev/vaadin/flow-components/vaadin-app-layout-flow-parent/vaadin-app-layout-flow-integration-tests
	at com.google.common.base.Preconditions.checkState(Preconditions.java:591)
	at org.openqa.selenium.remote.service.DriverService.checkExecutable(DriverService.java:146)
	at org.openqa.selenium.remote.service.DriverService.findExecutable(DriverService.java:141)
	at org.openqa.selenium.chrome.ChromeDriverService.access$000(ChromeDriverService.java:35)
	at org.openqa.selenium.chrome.ChromeDriverService$Builder.findDefaultExecutable(ChromeDriverService.java:159)
	at org.openqa.selenium.remote.service.DriverService$Builder.build(DriverService.java:355)
	at org.openqa.selenium.chrome.ChromeDriverService.createDefaultService(ChromeDriverService.java:94)
	at org.openqa.selenium.chrome.ChromeDriver.<init>(ChromeDriver.java:157)
	at com.vaadin.flow.testutil.ChromeBrowserTest.createHeadlessChromeDriver(ChromeBrowserTest.java:90)
	at com.vaadin.flow.testutil.ChromeBrowserTest.setup(ChromeBrowserTest.java:64)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.RunBefores.invokeMethod(RunBefores.java:33)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24)
	at com.vaadin.testbench.parallel.ParallelRunner$1.evaluate(ParallelRunner.java:464)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at com.vaadin.testbench.parallel.ParallelScheduler$1.call(ParallelScheduler.java:44)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
```

I was only able to get it around by using `AbstractParallelTest`. It seemed that the reason was somewhat that `AbstractComponentIT` doesn't work as soon as there is any `AbstractParallelTest` test near.

Reply:
You have to be careful here, there are two `AbstractComponentIT` classes. You want to import `com.vaadin.tests.AbstractComponentIT`. With the following code I'm not getting the error above:
```
import com.vaadin.tests.AbstractComponentIT;

@TestPath("vaadin-app-layout/i18n")
public class AppLayoutI18nIT extends AbstractComponentIT {
    private AppLayoutElement layout;

    @Before
    public void init() {
        open();
        layout = $(AppLayoutElement.class).waitForFirst();
    }
```

However I'm getting timeout errors when running the tests. When opening the test page in the browser it complains about different version of vaadin-lumo-styles:
```
Uncaught (in promise) DOMException: CustomElementRegistry.define: 'vaadin-lumo-styles' has already been defined as a custom element
```
That's probably because you upgraded the version of vaadin-app-layout. Not sure why we are still getting this error, I thought we do something special for pre-release versions to avoid that error. Maybe it's enough to rebase on master now that https://github.com/vaadin/flow-components/pull/2440 is merged.

Reply:
> You have to be careful here, there are two AbstractComponentIT classes. You want to import com.vaadin.tests.AbstractComponentIT. With the following code I'm not getting the error above:

You were right, I imported a different `AbstractComponentIT`. Now that I have the correct import, it works fine.

> Uncaught (in promise) DOMException: CustomElementRegistry.define: 'vaadin-lumo-styles' has already been defined as a custom element

Yeah, I saw this error too. I rebased the PR from `master` and the problem has gone.